<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">ATAC Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/ATAC/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="ATAC Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/ATAC/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/ATAC/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/ATAC/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.d7a12008.js" as="script">
<link rel="preload" href="/warp/assets/js/main.7a5a9da0.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">ATAC</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/ATAC/README">ATAC Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Multiome scATAC and GEX</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">RNA with UMIs</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Slide-seq</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Ultima Genomics Whole Genome Germline</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">WDL Best Practices</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>ATAC Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">1.0.0</a></td><td align="center">May, 2023</td><td align="center">Kaylee Mathews</td><td align="center">Please file GitHub issues in warp or contact <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">the WARP team</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-atac-workflow"></a>Introduction to the ATAC workflow<a class="hash-link" href="#introduction-to-the-atac-workflow" title="Direct link to heading">#</a></h2><p>ATAC is an open-source, cloud-optimized pipeline developed collaboration with members of the <a href="https://braininitiative.nih.gov/" target="_blank" rel="noopener noreferrer">BRAIN Initiative</a> (BICCN and <a href="https://brainblog.nih.gov/brain-blog/brain-issues-suite-funding-opportunities-advance-brain-cell-atlases-through-centers" target="_blank" rel="noopener noreferrer">BICAN</a> Sequencing Working Group) and <a href="https://nida.nih.gov/about-nida/organization/divisions/division-neuroscience-behavior-dnb/basic-research-hiv-substance-use-disorder/scorch-program" target="_blank" rel="noopener noreferrer">SCORCH</a> (see <a href="#acknowledgements">Acknowledgements</a> below). It supports the processing of 10x single-nucleus data generated with 10x Multiome <a href="https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression" target="_blank" rel="noopener noreferrer">ATAC-seq (Assay for Transposase-Accessible Chromatin)</a>, a technique used in molecular biology to assess genome-wide chromatin accessibility. </p><p>This workflow is the ATAC component of the <a href="/warp/docs/Pipelines/Multiome_Pipeline/README">Mutiome wrapper workflow</a>. It corrects cell barcodes (CBs), aligns reads to the genome, and produces a fragment file as well as per barcode metrics. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="quickstart-table"></a>Quickstart table<a class="hash-link" href="#quickstart-table" title="Direct link to heading">#</a></h2><p>The following table provides a quick glance at the ATAC pipeline features:</p><table><thead><tr><th>Pipeline features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay type</td><td>10x single cell or single nucleus ATAC</td><td><a href="https://www.10xgenomics.com" target="_blank" rel="noopener noreferrer">10x Genomics</a></td></tr><tr><td>Overall workflow</td><td>Barcode correction, read alignment, and fragment quanitification</td><td></td></tr><tr><td>Workflow language</td><td>WDL 1.0</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Genomic Reference Sequence</td><td>GRCh38 human genome primary sequence</td><td>GENCODE</td></tr><tr><td>Aligner</td><td>BWA-mem</td><td><a href="http://www.ncbi.nlm.nih.gov/pubmed/19451168" target="_blank" rel="noopener noreferrer">Li H. and Durbin R., 2009</a></td></tr><tr><td>Fragment quantification</td><td>SnapATAC2</td><td><a href="https://pubmed.ncbi.nlm.nih.gov/34774128/" target="_blank" rel="noopener noreferrer">Zhang, K. et al., 2021</a></td></tr><tr><td>Data input file format</td><td>File format in which sequencing data is provided</td><td><a href="https://academic.oup.com/nar/article/38/6/1767/3112533" target="_blank" rel="noopener noreferrer">FASTQ</a></td></tr><tr><td>Data output file format</td><td>File formats in which ATAC output is provided</td><td>TSV, h5ad, BAM</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="atac-installation"></a>ATAC installation<a class="hash-link" href="#atac-installation" title="Direct link to heading">#</a></h3><p>To download the latest ATAC release, see the release tags prefixed with &quot;Multiome&quot; on the WARP <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">releases page</a>. All ATAC pipeline releases are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/atac.changelog.md" target="_blank" rel="noopener noreferrer">ATAC changelog</a>. </p><p>To discover and search releases, use the WARP command-line tool <a href="https://github.com/broadinstitute/warp/tree/master/wreleaser" target="_blank" rel="noopener noreferrer">Wreleaser</a>.</p><p>ATAC can be deployed using <a href="https://cromwell.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. The workflow can also be run in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="input-variables"></a>Input Variables<a class="hash-link" href="#input-variables" title="Direct link to heading">#</a></h2><p>The following describes the inputs of the ATAC workflow. For more details on how default inputs are set for the Multiome workflow, see the <a href="/warp/docs/Pipelines/Multiome_Pipeline/README">Multiome overview</a>.</p><table><thead><tr><th>Variable name</th><th>Description</th></tr></thead><tbody><tr><td>read1_fastq_gzipped</td><td>Fastq inputs (array of compressed read 1 FASTQ files).</td></tr><tr><td>read2_fastq_gzipped</td><td>Fastq inputs (array of compressed read 2 FASTQ files containing cellular barcodes).</td></tr><tr><td>read3_fastq_gzipped</td><td>Fastq inputs (array of compressed read 3 FASTQ files).</td></tr><tr><td>output_base_name</td><td>Output prefix/base name for all intermediate files and pipeline outputs.</td></tr><tr><td>tar_bwa_reference</td><td>BWA reference (tar file containing reference fasta and corresponding files).</td></tr><tr><td>atac_gtf</td><td>CreateFragmentFile input variable: GTF file for SnapATAC2 to calculate TSS sites of fragment file.</td></tr><tr><td>chrom_sizes</td><td>CreateFragmentFile input variable: Text file containing chrom_sizes for genome build (i.e., hg38)</td></tr><tr><td>whitelist</td><td>Whitelist file for ATAC cellular barcodes.</td></tr><tr><td>adapter_seq_read1</td><td>TrimAdapters input: Sequence adapter for read 1 fastq.</td></tr><tr><td>adapter_seq_read3</td><td>TrimAdapters input: Sequence adapter for read 3 fastq.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="atac-tasks-and-tools"></a>ATAC tasks and tools<a class="hash-link" href="#atac-tasks-and-tools" title="Direct link to heading">#</a></h2><p>Overall, the ATAC workflow:</p><ol><li>Corrects CBs and partitions FASTQs by CB.</li><li>Aligns reads.</li><li>Merges aligned BAMs</li><li>Generates a fragment file</li><li>Calculates per cell barcode fragment metrics.</li></ol><p>The tools each ATAC task employs are detailed in the table below. </p><p>To see specific tool parameters, select the task WDL link in the table; then view the <code>command {}</code> section of the task in the WDL script. To view or use the exact tool software, see the task&#x27;s Docker image which is specified in the task WDL <code># runtime values</code> section as <code>String docker =</code>.</p><table><thead><tr><th>Task name and WDL link</th><th>Tool</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/skylab/FastqProcessing.wdl" target="_blank" rel="noopener noreferrer">FastqProcessing as SplitFastq</a></td><td>fastqprocess</td><td>custom</td><td>Corrects cell barcodes and splits fastqs into 30 GB FASTQs that are grouped by cell barcode with each read having the corrected (CB) and raw barcode (CR) in the read name.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/atac.wdl" target="_blank" rel="noopener noreferrer">TrimAdapters</a></td><td>Cutadapt v4.4</td><td>cutadapt</td><td>Trims adaptor sequences.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/atac.wdl" target="_blank" rel="noopener noreferrer">BWAPairedEndAlignment</a></td><td>bwa-mem2</td><td>mem</td><td>Aligns reads from each set of partitioned FASTQ files to the genome and outputs a BAM with ATAC barcodes in the CB:Z tag.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/tasks/skylab/MergeSortBam.wdl" target="_blank" rel="noopener noreferrer">Merge.MergeSortBamFiles as MergeBam</a></td><td>MergeSamFiles</td><td>Picard</td><td>Merges each BAM into a final aligned BAM with corrected cell barcodes in the CB tag.</td></tr><tr><td><a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/atac.wdl" target="_blank" rel="noopener noreferrer">CreateFragmentFile</a></td><td>make_fragment_file, import_data</td><td>SnapATAC2</td><td>Generates a fragment file from the final aligned BAM and outputs per barcode quality metrics in h5ad.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="output-variables"></a>Output variables<a class="hash-link" href="#output-variables" title="Direct link to heading">#</a></h2><table><thead><tr><th>Output variable name</th><th>Filename, if applicable</th><th>Output format and description</th></tr></thead><tbody><tr><td>bam_aligned_output</td><td><code>&lt;input_id&gt;</code>.bam</td><td>BAM containing aligned reads from ATAC workflow.</td></tr><tr><td>fragment_file</td><td><code>&lt;input_id&gt;</code>.fragments.tsv</td><td>TSV containing fragment start and stop coordinates per barcode.</td></tr><tr><td>snap_metrics</td><td><code>&lt;input_id</code>.metrics.h5ad</td><td>h5ad (Anndata) containing per barcode metrics from SnapATAC2.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="acknowledgements"></a>Acknowledgements<a class="hash-link" href="#acknowledgements" title="Direct link to heading">#</a></h2><p>We are immensely grateful to the members of the BRAIN Initiative (BICAN Sequencing Working Group) and SCORCH for their invaluable and exceptional contributions to this pipeline. Our heartfelt appreciation goes to Alex Dobin, Aparna Bhaduri, Alec Wysoker, Anish Chakka, Brian Herb, Daofeng Li, Fenna Krienen, Guo-Long Zuo, Jeff Goldy, Kai Zhang, Khalid Shakir, Bo Li, Mariano Gabitto, Michael DeBerardine, Mengyi Song, Melissa Goldman, Nelson Johansen, James Nemesh, and Theresa Hodges for their unwavering dedication and remarkable efforts.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/ATAC/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2023-06-26T14:14:29.000Z">6/26/2023</time></b> by <b>ekiernan</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/About_WARP/VersionAndReleasePipelines"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Version and Release Pipelines</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/CEMBA_MethylC_Seq_Pipeline/README"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">CEMBA Overview Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-atac-workflow" class="table-of-contents__link">Introduction to the ATAC workflow</a></li><li><a href="#quickstart-table" class="table-of-contents__link">Quickstart table</a></li><li><a href="#set-up" class="table-of-contents__link">Set-up</a><ul><li><a href="#atac-installation" class="table-of-contents__link">ATAC installation</a></li></ul></li><li><a href="#input-variables" class="table-of-contents__link">Input Variables</a></li><li><a href="#atac-tasks-and-tools" class="table-of-contents__link">ATAC tasks and tools</a></li><li><a href="#output-variables" class="table-of-contents__link">Output variables</a></li><li><a href="#acknowledgements" class="table-of-contents__link">Acknowledgements</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2023 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.d7a12008.js"></script>
<script src="/warp/assets/js/main.7a5a9da0.js"></script>
</body>
</html>