<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Smart-seq2 Single Nucleus Multi-Sample Count Matrix Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Smart-seq2 Single Nucleus Multi-Sample Count Matrix Overview | WARP"><meta data-react-helmet="true" name="description" content="The Smart-seq2 Single Nucleus Multi-Sample (Multi-snSS2) pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using Loompy v.3.0.6. It contains the raw cell-by-gene intron and exon counts."><meta data-react-helmet="true" property="og:description" content="The Smart-seq2 Single Nucleus Multi-Sample (Multi-snSS2) pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using Loompy v.3.0.6. It contains the raw cell-by-gene intron and exon counts."><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.ecbe2121.js" as="script">
<link rel="preload" href="/warp/assets/js/main.cc387073.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Multiome scATAC and GEX</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">RNA with UMIs</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Slide-seq</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/README">Smart-seq2 Single Nucleus Multi-Sample Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods">Smart-seq2 Single Nucleus Multi-Sample v1.2.0 Publication Methods</a></li><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview">Smart-seq2 Single Nucleus Multi-Sample Count Matrix Overview</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/consortia-processing">Consortia Data Processing</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Ultima Genomics Whole Genome Germline</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">WDL Best Practices</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Smart-seq2 Single Nucleus Multi-Sample Count Matrix Overview</h1></header><p>The Smart-seq2 Single Nucleus Multi-Sample (Multi-snSS2) pipeline&#x27;s default count matrix output is a Loom file, an HDF5 file generated using <a href="http://loompy.org/" target="_blank" rel="noopener noreferrer">Loompy v.3.0.6</a>. It contains the raw cell-by-gene intron and exon counts.</p><p>The matrix also contains multiple metrics for both individual cells (the columns of the matrix; <a href="#table-2-column-attributes-cell-metrics">Table 2</a>) and individual genes (the rows of the matrix; <a href="#table-3-row-attributes-gene-metrics">Table 3</a>).</p><p>Additional details for each metric are provided in the JAVA source code for Picard&#x27;s <a href="https://github.com/broadinstitute/picard/blob/4527d6de2f33f98f80e32e3acd32b09633529bd0/src/main/java/picard/analysis/AlignmentSummaryMetrics.java" target="_blank" rel="noopener noreferrer">AlignmentSummaryMetrics</a>, <a href="https://github.com/broadinstitute/picard/blob/4527d6de2f33f98f80e32e3acd32b09633529bd0/src/main/java/picard/analysis/GcBiasSummaryMetrics.java" target="_blank" rel="noopener noreferrer">GcBiasSummaryMetrics</a>, and <a href="https://github.com/broadinstitute/picard/blob/4527d6de2f33f98f80e32e3acd32b09633529bd0/src/main/java/picard/sam/DuplicationMetrics.java" target="_blank" rel="noopener noreferrer">DuplicationMetrics</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-1-global-attributes"></a>Table 1. Global attributes<a class="hash-link" href="#table-1-global-attributes" title="Direct link to heading">#</a></h2><p>The global attributes in the Loom apply to the whole file, not any specific part. </p><table><thead><tr><th align="left">Attribute</th><th align="left">Details</th></tr></thead><tbody><tr><td align="left"><code>CreationDate</code></td><td align="left">Date the Loom file was created.</td></tr><tr><td align="left"><code>LOOM_SPEC_VERSION</code></td><td align="left">Loom file spec version used during creation of the Loom file.</td></tr><tr><td align="left"><code>batch_id</code></td><td align="left">The <code>batch_id</code> provided to the pipeline as input.</td></tr><tr><td align="left"><code>pipeline_version</code></td><td align="left">Version of the Multi-snSS2 pipeline used to generate the Loom file.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-2-column-attributes-cell-metrics"></a>Table 2. Column attributes (cell metrics)<a class="hash-link" href="#table-2-column-attributes-cell-metrics" title="Direct link to heading">#</a></h2><p>The cell metrics below are computed using <a href="https://broadinstitute.github.io/picard/" target="_blank" rel="noopener noreferrer">Picard</a>, with the exception of <code>CellID</code>, <code>cell_names</code>, and <code>input_id</code> which are provided to the pipeline as input.</p><table><thead><tr><th align="left">Cell Metrics</th><th align="left">Tool</th><th align="left">Details</th></tr></thead><tbody><tr><td align="left"><code>ACCUMULATION_LEVEL</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832753924763" target="_blank" rel="noopener noreferrer">CollectMultipleMetrics</a></td><td align="left">Level of metric accumulation; set to <code>ALL_READS</code> using the <code>--METRIC_ACCUMULATION_LEVEL</code> argument.</td></tr><tr><td align="left"><code>ALIGNED_READS</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Total number of aligned reads produced in a run.</td></tr><tr><td align="left"><code>AT_DROPOUT</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Percentage of misaligned reads with GC content below 50%.</td></tr><tr><td align="left"><code>AVG_POS_3PRIME_SOFTCLIP_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Average length of soft-clipped bases at the 3&#x27; end of the first reads.</td></tr><tr><td align="left"><code>AVG_POS_3PRIME_SOFTCLIP_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Average length of soft-clipped bases at the 3&#x27; end of all reads.</td></tr><tr><td align="left"><code>AVG_POS_3PRIME_SOFTCLIP_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Average length of soft-clipped bases at the 3&#x27; end of the second reads.</td></tr><tr><td align="left"><code>BAD_CYCLES.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of cycles with combined no-call and mismatch rates greater than or equal to 80% for the first reads.</td></tr><tr><td align="left"><code>BAD_CYCLES.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of cycles with combined no-call and mismatch rates greater than or equal to 80% for all reads.</td></tr><tr><td align="left"><code>BAD_CYCLES.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of cycles with combined no-call and mismatch rates greater than or equal to 80% for the second reads.</td></tr><tr><td align="left"><code>CellID</code></td><td align="left"><a href="https://github.com/broadinstitute/warp-tools" target="_blank" rel="noopener noreferrer">warp-tools</a></td><td align="left">Unique identifier for each cell provided to the pipeline as <code>input_ids</code>; identical to <code>cell_names</code> and <code>input_id</code>.</td></tr><tr><td align="left"><code>ESTIMATED_LIBRARY_SIZE</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Estimated number of unique molecules in the library based on paired-end duplication.</td></tr><tr><td align="left"><code>GC_DROPOUT</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Percentage of misaligned reads with GC content above 50%.</td></tr><tr><td align="left"><code>GC_NC_0_19</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Normalized coverage over reads with GC content from 0 - 19%.</td></tr><tr><td align="left"><code>GC_NC_20_39</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Normalized coverage over reads with GC content from 20 - 39%.</td></tr><tr><td align="left"><code>GC_NC_40_59</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Normalized coverage over reads with GC content from 40 - 59%.</td></tr><tr><td align="left"><code>GC_NC_60_79</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Normalized coverage over reads with GC content from 60 - 79%.</td></tr><tr><td align="left"><code>GC_NC_80_100</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Normalized coverage over reads with GC content from 80 - 100%.</td></tr><tr><td align="left"><code>MAD_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median absolute deviation of the lengths of forward reads.</td></tr><tr><td align="left"><code>MAD_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median absolute deviation of the lengths of all reads.</td></tr><tr><td align="left"><code>MAD_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median absolute deviation of the lengths of reverse reads.</td></tr><tr><td align="left"><code>MAX_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Maximum length of forward reads.</td></tr><tr><td align="left"><code>MAX_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Maximum length of all reads.</td></tr><tr><td align="left"><code>MAX_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Maximum length of reverse reads.</td></tr><tr><td align="left"><code>MEAN_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Mean length of forward reads.</td></tr><tr><td align="left"><code>MEAN_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Mean length of all reads.</td></tr><tr><td align="left"><code>MEAN_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Mean length of reverse reads.</td></tr><tr><td align="left"><code>MEDIAN_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median length of forward reads.</td></tr><tr><td align="left"><code>MEDIAN_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median length of all reads.</td></tr><tr><td align="left"><code>MEDIAN_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median length of reverse reads.</td></tr><tr><td align="left"><code>MIN_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Minimum length of forward reads.</td></tr><tr><td align="left"><code>MIN_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Minimum length of all reads.</td></tr><tr><td align="left"><code>MIN_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Minimum length of reverse reads.</td></tr><tr><td align="left"><code>PCT_ADAPTER.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter forward reads that are unaligned or aligned with a mapping quality of 0 and match to a known adapter sequence from the start of the read.</td></tr><tr><td align="left"><code>PCT_ADAPTER.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of all pass-filter reads that are unaligned or aligned with a mapping quality of 0 and match to a known adapter sequence from the start of the read.</td></tr><tr><td align="left"><code>PCT_ADAPTER.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter reverse reads that are unaligned or aligned with a mapping quality of 0 and match to a known adapter sequence from the start of the read.</td></tr><tr><td align="left"><code>PCT_CHIMERAS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of forward reads where the insert is larger than 100 kb or the ends of the pair map to different chromosomes.</td></tr><tr><td align="left"><code>PCT_CHIMERAS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of all reads where the insert is larger than 100 kb or the ends of the pair map to different chromosomes.</td></tr><tr><td align="left"><code>PCT_CHIMERAS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reverse reads where the insert is larger than 100 kb or the ends of the pair map to different chromosomes.</td></tr><tr><td align="left"><code>PCT_HARDCLIP.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are hard-clipped from aligned, forward reads.</td></tr><tr><td align="left"><code>PCT_HARDCLIP.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are hard-clipped from all aligned reads.</td></tr><tr><td align="left"><code>PCT_HARDCLIP.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are hard-clipped from aligned, reverse reads.</td></tr><tr><td align="left"><code>PCT_PF_READS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of forward reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PCT_PF_READS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PCT_PF_READS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reverse reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PCT_PF_READS_ALIGNED.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter forward reads that are aligned.</td></tr><tr><td align="left"><code>PCT_PF_READS_ALIGNED.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of all pass-filter reads that are aligned.</td></tr><tr><td align="left"><code>PCT_PF_READS_ALIGNED.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter reverse reads that are aligned.</td></tr><tr><td align="left"><code>PCT_PF_READS_IMPROPER_PAIRS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of forward reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_PF_READS_IMPROPER_PAIRS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_PF_READS_IMPROPER_PAIRS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reverse reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_READS_ALIGNED_IN_PAIRS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of forward reads properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_READS_ALIGNED_IN_PAIRS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reads properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_READS_ALIGNED_IN_PAIRS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of reverse reads properly aligned in pairs.</td></tr><tr><td align="left"><code>PCT_SOFTCLIP.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are soft-clipped from aligned forward reads.</td></tr><tr><td align="left"><code>PCT_SOFTCLIP.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are soft-clipped from all aligned reads.</td></tr><tr><td align="left"><code>PCT_SOFTCLIP.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of pass-filter bases that are soft-clipped from aligned reverse reads.</td></tr><tr><td align="left"><code>PERCENT_DUPLICATION</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Fraction of mapped sequence marked as duplicate.</td></tr><tr><td align="left"><code>PF_ALIGNED_BASES.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of aligned bases in pass-filter forward reads.</td></tr><tr><td align="left"><code>PF_ALIGNED_BASES.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of aligned bases in all pass-filter reads.</td></tr><tr><td align="left"><code>PF_ALIGNED_BASES.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of aligned bases in pass-filter reverse reads.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_BASES.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of bases aligned to the reference sequence in forward reads with high mapping quality.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_BASES.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of bases aligned to the reference sequence in all reads with high mapping quality.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_BASES.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of bases aligned to the reference sequence in reverse reads with high mapping quality.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_Q20_BASES.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Subset of <code>PF_HQ_ALIGNED_BASES.FIRST_OF_PAIR</code> with a base call quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_Q20_BASES.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Subset of <code>PF_HQ_ALIGNED_BASES.PAIR</code> with a base call quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_Q20_BASES.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Subset of <code>PF_HQ_ALIGNED_BASES.SECOND_OF_PAIR</code> with a base call quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_READS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter forward reads aligned with a mapping quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_READS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of all pass-filter reads aligned with a mapping quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ALIGNED_READS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter reverse reads aligned with a mapping quality of at least 20.</td></tr><tr><td align="left"><code>PF_HQ_ERROR_RATE.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of bases in pass-filter, high-quality forward reads that do not match the reference.</td></tr><tr><td align="left"><code>PF_HQ_ERROR_RATE.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of bases in all pass-filter, high-quality reads that do not match the reference.</td></tr><tr><td align="left"><code>PF_HQ_ERROR_RATE.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Fraction of bases in pass-filter, high-quality reverse reads that do not match the reference.</td></tr><tr><td align="left"><code>PF_HQ_MEDIAN_MISMATCHES.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median number of mismatches in high-quality forward reads.</td></tr><tr><td align="left"><code>PF_HQ_MEDIAN_MISMATCHES.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median number of mismatches in all high-quality reads.</td></tr><tr><td align="left"><code>PF_HQ_MEDIAN_MISMATCHES.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Median number of mismatches in high-quality reverse reads.</td></tr><tr><td align="left"><code>PF_INDEL_RATE.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of insertion and deletion events per 100 aligned bases in forward reads.</td></tr><tr><td align="left"><code>PF_INDEL_RATE.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of insertion and deletion events per 100 aligned bases in all reads.</td></tr><tr><td align="left"><code>PF_INDEL_RATE.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of insertion and deletion events per 100 aligned bases in reverse reads.</td></tr><tr><td align="left"><code>PF_MISMATCH_RATE.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Rate of base mismatching for all aligned bases in forward reads.</td></tr><tr><td align="left"><code>PF_MISMATCH_RATE.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Rate of base mismatching for all aligned bases in all reads.</td></tr><tr><td align="left"><code>PF_MISMATCH_RATE.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Rate of base mismatching for all aligned bases in reverse reads.</td></tr><tr><td align="left"><code>PF_NOISE_READS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter forward reads marked as noise.</td></tr><tr><td align="left"><code>PF_NOISE_READS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of all pass-filter reads marked as noise.</td></tr><tr><td align="left"><code>PF_NOISE_READS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter reverse reads marked as noise.</td></tr><tr><td align="left"><code>PF_READS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of forward reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PF_READS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PF_READS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reverse reads that pass vendor check (pass-filter).</td></tr><tr><td align="left"><code>PF_READS_ALIGNED.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter forward reads that are aligned.</td></tr><tr><td align="left"><code>PF_READS_ALIGNED.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter reads that are aligned.</td></tr><tr><td align="left"><code>PF_READS_ALIGNED.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter reverse reads that are aligned.</td></tr><tr><td align="left"><code>PF_READS_IMPROPER_PAIRS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of forward reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>PF_READS_IMPROPER_PAIRS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>PF_READS_IMPROPER_PAIRS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reverse reads not properly aligned in pairs.</td></tr><tr><td align="left"><code>READS_ALIGNED_IN_PAIRS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of forward reads properly aligned in pairs.</td></tr><tr><td align="left"><code>READS_ALIGNED_IN_PAIRS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reads properly aligned in pairs.</td></tr><tr><td align="left"><code>READS_ALIGNED_IN_PAIRS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of reverse reads properly aligned in pairs.</td></tr><tr><td align="left"><code>READS_USED</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">String describing whether duplicates are included in metrics produced by <a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a>; the pipeline removes duplicates before metrics are calculated.</td></tr><tr><td align="left"><code>READ_PAIRS_EXAMINED</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of mapped read pairs examined by <a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a>.</td></tr><tr><td align="left"><code>READ_PAIR_DUPLICATES</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of read pairs marked as duplicates.</td></tr><tr><td align="left"><code>READ_PAIR_OPTICAL_DUPLICATES</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of read pairs duplicates caused by optical duplication.</td></tr><tr><td align="left"><code>SD_READ_LENGTH.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Standard deviation of forward read lengths.</td></tr><tr><td align="left"><code>SD_READ_LENGTH.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Standard deviation of read lengths.</td></tr><tr><td align="left"><code>SD_READ_LENGTH.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Standard deviation of reverse read lengths.</td></tr><tr><td align="left"><code>SECONDARY_OR_SUPPLEMENTARY_RDS</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of secondary or supplemetary reads.</td></tr><tr><td align="left"><code>STRAND_BALANCE.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter forward reads aligned divided by the total number of pass-filter reads aligned.</td></tr><tr><td align="left"><code>STRAND_BALANCE.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Average strand balance of forward and reverse reads.</td></tr><tr><td align="left"><code>STRAND_BALANCE.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Number of pass-filter reverse reads aligned divided by the total number of pass-filter reads aligned.</td></tr><tr><td align="left"><code>TOTAL_CLUSTERS</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Total number of reads after filtering used in GC bias calculation.</td></tr><tr><td align="left"><code>TOTAL_READS.FIRST_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of forward reads.</td></tr><tr><td align="left"><code>TOTAL_READS.PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of reads.</td></tr><tr><td align="left"><code>TOTAL_READS.SECOND_OF_PAIR</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832683660955" target="_blank" rel="noopener noreferrer">CollectAlignmentSummaryMetrics</a></td><td align="left">Total number of reverse reads.</td></tr><tr><td align="left"><code>UNMAPPED_READS</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of unmapped reads examined by <a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a>.</td></tr><tr><td align="left"><code>UNPAIRED_READS_EXAMINED</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of mapped reads without a mapped mate pair examined by <a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a>.</td></tr><tr><td align="left"><code>UNPAIRED_READ_DUPLICATES</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832748517275" target="_blank" rel="noopener noreferrer">MarkDuplicates</a></td><td align="left">Number of fragments marked as duplicates.</td></tr><tr><td align="left"><code>WINDOW_SIZE</code></td><td align="left"><a href="https://gatk.broadinstitute.org/hc/en-us/articles/13832711132315" target="_blank" rel="noopener noreferrer">CollectGcBiasMetrics</a></td><td align="left">Genomic window size used in GC bias calculation.</td></tr><tr><td align="left"><code>cell_names</code></td><td align="left"><a href="https://github.com/broadinstitute/warp-tools" target="_blank" rel="noopener noreferrer">warp-tools</a></td><td align="left">Unique identifier for each cell provided to the pipeline as <code>input_ids</code>; identical to <code>Cell_ID</code> and <code>input_id</code>.</td></tr><tr><td align="left"><code>input_id</code></td><td align="left"><a href="https://github.com/broadinstitute/warp-tools" target="_blank" rel="noopener noreferrer">warp-tools</a></td><td align="left">Unique identifier for each cell provided to the pipeline as <code>input_ids</code>; identical to <code>Cell_ID</code> and <code>cell_names</code>.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="table-3-row-attributes-gene-metrics"></a>Table 3. Row attributes (gene metrics)<a class="hash-link" href="#table-3-row-attributes-gene-metrics" title="Direct link to heading">#</a></h2><table><thead><tr><th>Gene Metrics</th><th>Tool</th><th>Details</th></tr></thead><tbody><tr><td><code>Gene</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The unique <code>gene_ids</code> provided in the <a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a>; identical to the <code>ensembl_ids</code> attribute.</td></tr><tr><td><code>ensembl_ids</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The unique <code>gene_ids</code> provided in the <a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a>; identical to the <code>Gene</code> attribute.</td></tr><tr><td><code>exon_lengths</code></td><td><a href="https://github.com/broadinstitute/warp-tools" target="_blank" rel="noopener noreferrer">warp-tools</a></td><td>The length in base pairs of the exons corresponding to this entity.</td></tr><tr><td><code>gene_names</code></td><td><a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a></td><td>The unique <code>gene_name</code> provided in the <a href="https://www.gencodegenes.org/" target="_blank" rel="noopener noreferrer">GENCODE GTF</a>.</td></tr><tr><td><code>intron_lengths</code></td><td><a href="https://github.com/broadinstitute/warp-tools" target="_blank" rel="noopener noreferrer">warp-tools</a></td><td>The length in base pairs of the introns corresponding to this entity.</td></tr></tbody></table></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/count-matrix-overview.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2023-06-27T15:18:14.000Z">6/27/2023</time></b> by <b>ekiernan</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/multi_snss2.methods"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Smart-seq2 Single Nucleus Multi-Sample v1.2.0 Publication Methods</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Smart-seq2_Single_Nucleus_Multi_Sample_Pipeline/consortia-processing"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Consortia Data Processing Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#table-1-global-attributes" class="table-of-contents__link">Table 1. Global attributes</a></li><li><a href="#table-2-column-attributes-cell-metrics" class="table-of-contents__link">Table 2. Column attributes (cell metrics)</a></li><li><a href="#table-3-row-attributes-gene-metrics" class="table-of-contents__link">Table 3. Row attributes (gene metrics)</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2023 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.ecbe2121.js"></script>
<script src="/warp/assets/js/main.cc387073.js"></script>
</body>
</html>