<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">WDL cost optimization: Tips and tricks when working with Google Cloud in Terra | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Best_practices/GC_cost_optimization"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="WDL cost optimization: Tips and tricks when working with Google Cloud in Terra | WARP"><meta data-react-helmet="true" name="description" content="Reducing the cost of your WDL workflow is always a priority. Below, the Broad Institute’s Pipelines team provides some tips and tricks for optimizing workflow costs for runs using Google Cloud virtual machines (VM) from the bioinformatics platform, Terra."><meta data-react-helmet="true" property="og:description" content="Reducing the cost of your WDL workflow is always a priority. Below, the Broad Institute’s Pipelines team provides some tips and tricks for optimizing workflow costs for runs using Google Cloud virtual machines (VM) from the bioinformatics platform, Terra."><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Best_practices/GC_cost_optimization"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Best_practices/GC_cost_optimization" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Best_practices/GC_cost_optimization" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.5c082446.js" as="script">
<link rel="preload" href="/warp/assets/js/main.1d35face.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">🌜</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">🌞</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">WARP Pipelines</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WDL Best Practices</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Best_practices/GC_cost_optimization">WDL cost optimization: Tips and tricks when working with Google Cloud in Terra</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Best_practices/autosize">Autosizing disk for Google Cloud</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Best_practices/setting_defaults">Setting default values</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Best_practices/reusing_code">Reusing WDL code</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Best_practices/suggested_formats">WDL formatting tips</a></li><li class="menu__list-item"><a class="menu__link" tabindex="0" href="/warp/docs/Best_practices/task_execution">Task execution - tips for using the WDL task command section</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>WDL cost optimization: Tips and tricks when working with Google Cloud in Terra</h1></header><p>Reducing the cost of your WDL workflow is always a priority. Below, the Broad Institute’s Pipelines team provides some tips and tricks for optimizing workflow costs for runs using Google Cloud virtual machines (VM) from the bioinformatics platform, <a href="/warp/docs/Best_practices/app.terra.bio">Terra</a>. </p><p>Overall, the majority of optimization comes down to understanding the size of your VM and how long you use it. Keep this in mind as you read the tips below and remember: one size does not necessarily fit all when it comes to optimizing your workflow.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tip-1-know-the-difference-between-tools-and-bash-commands"></a>Tip 1: Know the difference between tools and bash commands<a class="hash-link" href="#tip-1-know-the-difference-between-tools-and-bash-commands" title="Direct link to heading">#</a></h2><p>A big part of cost reduction is understanding what tools do the heavy lifting in your analysis because these tools will often dictate the size requirements of a VM and how long it runs. When we talk about software tools, we mean the time and memory-intensive tools required to analyze your data. For example, if you’re doing sequencing alignment, your tools might be bwa, STAR, or whatever your favorite aligner is. </p><p>In contrast, bash commands are primarily used to help (re)format, (re)name, and move files around to get them ready to do the large analysis. These functions will likely have a smaller impact on the necessary VM size than your tools.</p><p>The distinction between tools and bash commands becomes important when we discuss running one tool per WDL task in Tip 2. Before reading on, think about what tools vs. bash operations you currently use for your analysis. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tip-2-modularize-tools-when-possible"></a>Tip 2: Modularize tools when possible<a class="hash-link" href="#tip-2-modularize-tools-when-possible" title="Direct link to heading">#</a></h2><p>When you run a WDL workflow in the cloud, you specify a docker for each task, which will be used to set up a virtual machine (VM). Initiating a VM with a docker comes with some overhead costs so it might be tempting to avoid these by running all of your software tools in a single task, using only a single VM and docker. </p><p>Unfortunately, the reality is that software tools aren’t 100% reliable, and when they fail, it means recreating a VM and rerunning your tools to successfully complete your workflow run. When you run multiple tools on a single VM or in a single task, it doesn’t matter if one tool fails or multiple tools fail, you’ll have to rerun everything that’s part of the WDL task. You’ll also have a hard time making updates to the code should you want to reuse one of the tools in a separate workflow later on.</p><p>For these reasons, we recommend modularizing workflows when possible, limiting the number of software tools you run per task. For example, if you&#x27;re software tool takes more than 10 minutes to run, it&#x27;s probably a good idea to put the tool in it&#x27;s own task. When you modularize tools, you’ll be able to save your upstream analysis even if your tool fails, saving you the cost of time inside your VM. You’ll also be able to better customize the VM to meet the needs of the tools and run more efficiently.</p><p>In addition to modularizing tools, you can also consider whether you want to modularize bash commands. Bash commands are often less time intensive because they are quick steps that you use to prepare files for processing with your software tool. In the case of quick bash commands, it makes sense to group the, in the same WDL task as you would your tool as opposed to separating them into their own task. </p><p>Let’s look at an example WDL task below where we first use bash commands to prepare sequencing data for genomic alignment with the STAR aligner, and then run the STAR tool:</p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly wdl"><pre tabindex="0" class="prism-code language-wdl codeBlock_23N8 thin-scrollbar" style="color:#bfc7d5;background-color:#292d3e"><code class="codeBlockLines_39YC"><span class="token-line" style="color:#bfc7d5"><span class="token plain">command &lt;&lt;&lt;</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  Set -e</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain"># prepare reference with bash</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">mkdir genome_reference</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">Tar -xf “~{tar_star_reference}” -C genome_reference --strip component 1</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">Rm “~{tar_star_reference}</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">STAR \</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">-- runMode alignReads \</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">--runThreadN ~{cpu}</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">--genomeDir genome_reference</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">&gt;&gt;&gt;</span></span></code></pre><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o clean-btn">Copy</button></div></div><p>In the use case above, it makes sense to combine the bash operations and STAR. The bash operations are only used to make a new directory for the genome references and to compress files. After the files are ready, the heavy lifting of genome alignment is done by STAR.</p><p>Keep in mind there might be scenarios where splitting out more intensive bash operations might make sense. For example, if your bash operations are time intensive and your tool is not very reliable, you might want to save the outputs of your bash functions before moving on to running the tool. In this case, it would make sense to keep the bash commands and your tool in separate WDL tasks to avoid rerunning them both.</p><p>Similarly, there are exceptions to modularization of tools. For example, if you&#x27;re working with a lot of intermediate files, it might make sense to string two tools together in a single task if it helps avoid costly egress. Always keep in mind timing and the reliability of your tools; remember that these tips are guidelines, not hard rules. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tip-3-avoid-inputoutput-timing-costs-moving-and-loading-lots-of-files"></a>Tip 3: Avoid input/output timing costs (moving and loading lots of files)<a class="hash-link" href="#tip-3-avoid-inputoutput-timing-costs-moving-and-loading-lots-of-files" title="Direct link to heading">#</a></h2><p>Just like starting a VM has some overhead cost, localizing files also has some downsides, including using more time in your VM. And the more time you spend, the more your cost will go up. Your VM needs to find all your cloud files in order to run them through your different tools. That means if you have thousands of files, you’ll have to run initialization steps for each of them, which can take a bit of time and networking. </p><p>Each time you move cloud files, you pay egress for the network transitions, so it’s important to find the balance in the number of files you decide to move. You have to weigh whether it costs more to move a large file vs. moving several smaller files. For example, you might find that it’s more cost-efficient to move a zipped ~100 GB file than to move 100, 1 GB files. After running a  test workflow, check your workflow logs to see what the timing is for localizing and moving files vs. running your tool. This might require trial and error when developing your workflow. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tip-4-run-files-in-parallel-when-possible"></a>Tip 4: Run files in parallel when possible<a class="hash-link" href="#tip-4-run-files-in-parallel-when-possible" title="Direct link to heading">#</a></h2><p>If you need to run multiple files through a tool, you’ll have to decide whether to scatter those files across multiple VMs (running the tool in parallel), or run the files sequentially through the tool in one VM. </p><p>Similar to the problem of running multiple tools per VM, running multiple files per VM also comes with the risk of rerunning your tool if a file should fail. If you’re running files that are large or prone to transient failures, it’s best to scatter them across VMs in parallel. </p><p>Let’s take a look at an example WDL script that shows scattering; it performs variant calling on large BAM files with a tool called HaplotypeCaller. The WDL scatters the input array of BAM files so that the task runs on each individual BAM. It creates a single VM for each instance of the task. </p><div class="codeBlockContainer_K1bP"><div class="codeBlockContent_hGly wdl"><pre tabindex="0" class="prism-code language-wdl codeBlock_23N8 thin-scrollbar" style="color:#bfc7d5;background-color:#292d3e"><code class="codeBlockLines_39YC"><span class="token-line" style="color:#bfc7d5"><span class="token plain">version 1.0</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">workflow ScatterGatherExample {</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  input {</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">    Array[File] sampleBAMs</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  }</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain" style="display:inline-block">
</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  scatter (sample in sampleBAMs) {</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">    call HaplotypeCallerERC { </span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">      input: </span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">        bamFile=sample </span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">    }</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  }</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  call GenotypeGVCF {</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">   input: </span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">     GVCFs = HaplotypeCallerERC.GVCF </span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">  }</span></span><span class="token-line" style="color:#bfc7d5"><span class="token plain">}</span></span></code></pre><button type="button" aria-label="Copy code to clipboard" class="copyButton_Ue-o clean-btn">Copy</button></div></div><p>If one of the BAM files in the array fails to run through HaplotypeCaller, the remaining files will still be stored. In consequence, you can rerun the WDL to analyze the failed file and not waste additional resources rerunning the files that already successfully ran through the task.</p><p>Another consideration when running files in parallel is to make sure that files are split into roughly equal size chunks. This is important if, for example, you have hard-coded values for memory and disk. In that case, each shard will get as much memory and disk as needed for your largest shard and the largest one can sometimes require a long time to run.</p><p>As with the other tips, running in parallel might not be necessary for all use cases. If you’re running smaller and fewer files, you can save money by running sequentially through the task in a single VM. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tip-5-use-a-vm-that-already-has-your-reference"></a>Tip 5: Use a VM that already has your reference<a class="hash-link" href="#tip-5-use-a-vm-that-already-has-your-reference" title="Direct link to heading">#</a></h2><p>When you run WDL workflows in Terra, you have the option to use a VM that already has reference files loaded. This is important because you can avoid localization costs of necessary reference files, saving you time in the VM and ultimately, money.</p><p>Let’s look at an example of how using preloaded references helped optimize timing for the alignment task of a real workflow, the <a href="https://github.com/broadinstitute/warp/blob/master/pipelines/skylab/smartseq2_single_nucleus_multisample/MultiSampleSmartSeq2SingleNucleus.wdl" target="_blank" rel="noopener noreferrer">Smart-seq2 Single Nucleus Multi-Sample Pipeline</a>. </p><p>Prior to using preloaded references, the timing for the alignment task using STAR was the following:</p><table><thead><tr><th>Timing of task steps</th><th>Task step (from log file)</th></tr></thead><tbody><tr><td>00:52:29</td><td>Starting container setup.</td></tr><tr><td>00:52:38</td><td>Localization script execution started.</td></tr><tr><td>00:52:45</td><td>Localizing input</td></tr><tr><td>01:03:29</td><td>Done localization.</td></tr><tr><td>01:03:30</td><td>Running user action: docker</td></tr><tr><td>01:17:14</td><td>..... started STAR run</td></tr><tr><td>01:17:15</td><td>..... loading genome</td></tr><tr><td>01:30:31</td><td>..... started mapping</td></tr><tr><td>01:32:25</td><td>..... finished mapping</td></tr><tr><td>01:32:27</td><td>..... started sorting BAM</td></tr><tr><td>01:32:29</td><td>..... finished successfully</td></tr><tr><td>01:32:41</td><td>Starting delocalization.</td></tr><tr><td>01:32:52</td><td>Done delocalization.</td></tr></tbody></table><p>Notice that the localizing input takes ~ 11 min and that loading the genome takes an additional 13 min. In contrast, mapping the files to the genome (the actual task) takes only ~ 3 min. This means we spend most of our time just finding and loading files. </p><p>In contrast, when we use preloaded references, the timing looks more like the following:</p><table><thead><tr><th>Timing of task steps</th><th>Task step (from log file)</th></tr></thead><tbody><tr><td>00:52:29</td><td>Starting container setup.</td></tr><tr><td>00:52:38</td><td>Localization script execution started.</td></tr><tr><td>00:52:45</td><td>Localizing input</td></tr><tr><td>00:52:55</td><td>Done localization.</td></tr></tbody></table><p>Notice it now only takes 10 seconds to localize the files! </p><p>When you’re troubleshooting workflows in Terra, be sure to check out your workflow timing logs so you can identify which steps are costing you time in the VM.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="conclusion"></a>Conclusion<a class="hash-link" href="#conclusion" title="Direct link to heading">#</a></h2><p>Workflow cost optimization is a bit of an art, but the payoff is worth the effort. Overall, remember that the size of your VM and the time you spend using it drive your cost. Use your job manager logs wisely to look for pain points, and modularize and parallelize when possible!</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Best_practices/GC_cost_optimization.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2023-01-13T14:03:57.000Z">1/13/2023</time></b> by <b>Jessica Way</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Whole_Genome_Germline_Single_Sample_Pipeline/wgs.methods"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">« Whole Genome Germline Single Sample v3.1.6 Methods (Default workflow)</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Best_practices/autosize"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Autosizing disk for Google Cloud »</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#tip-1-know-the-difference-between-tools-and-bash-commands" class="table-of-contents__link">Tip 1: Know the difference between tools and bash commands</a></li><li><a href="#tip-2-modularize-tools-when-possible" class="table-of-contents__link">Tip 2: Modularize tools when possible</a></li><li><a href="#tip-3-avoid-inputoutput-timing-costs-moving-and-loading-lots-of-files" class="table-of-contents__link">Tip 3: Avoid input/output timing costs (moving and loading lots of files)</a></li><li><a href="#tip-4-run-files-in-parallel-when-possible" class="table-of-contents__link">Tip 4: Run files in parallel when possible</a></li><li><a href="#tip-5-use-a-vm-that-already-has-your-reference" class="table-of-contents__link">Tip 5: Use a VM that already has your reference</a></li><li><a href="#conclusion" class="table-of-contents__link">Conclusion</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright © 2023 Copyright © Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.5c082446.js"></script>
<script src="/warp/assets/js/main.1d35face.js"></script>
</body>
</html>