<!doctype html>
<html lang="en" dir="ltr">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width,initial-scale=1">
<meta name="generator" content="Docusaurus v2.0.0-beta.5">
<link rel="alternate" type="application/rss+xml" href="/warp/blog/rss.xml" title="WARP Blog RSS Feed">
<link rel="alternate" type="application/atom+xml" href="/warp/blog/atom.xml" title="WARP Blog Atom Feed">
<link rel="preconnect" href="https://www.google-analytics.com">
<script>window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)},ga.l=+new Date,ga("create","UA-181048107-1","auto"),ga("set","anonymizeIp",!0),ga("send","pageview")</script>
<script async src="https://www.google-analytics.com/analytics.js"></script>
<link rel="search" type="application/opensearchdescription+xml" title="WARP" href="/warp/opensearch.xml">
<script>!function(t,h,e,j,s,n){t.hj=t.hj||function(){(t.hj.q=t.hj.q||[]).push(arguments)},t._hjSettings={hjid:2427684,hjsv:6},s=h.getElementsByTagName("head")[0],(n=h.createElement("script")).async=1,n.src="https://static.hotjar.com/c/hotjar-"+t._hjSettings.hjid+".js?sv="+t._hjSettings.hjsv,s.appendChild(n)}(window,document)</script><title data-react-helmet="true">Multiome Overview | WARP</title><meta data-react-helmet="true" name="twitter:card" content="summary_large_image"><meta data-react-helmet="true" property="og:url" content="https://broadinstitute.github.io/warp/docs/Pipelines/Multiome_Pipeline/README"><meta data-react-helmet="true" name="docsearch:language" content="en"><meta data-react-helmet="true" name="docsearch:version" content="current"><meta data-react-helmet="true" name="docsearch:docusaurus_tag" content="docs-default-current"><meta data-react-helmet="true" property="og:title" content="Multiome Overview | WARP"><meta data-react-helmet="true" name="description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><meta data-react-helmet="true" property="og:description" content="| Pipeline Version | Date Updated | Documentation Author | Questions or Feedback |"><link data-react-helmet="true" rel="shortcut icon" href="/warp/img/favicon.ico"><link data-react-helmet="true" rel="canonical" href="https://broadinstitute.github.io/warp/docs/Pipelines/Multiome_Pipeline/README"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Multiome_Pipeline/README" hreflang="en"><link data-react-helmet="true" rel="alternate" href="https://broadinstitute.github.io/warp/docs/Pipelines/Multiome_Pipeline/README" hreflang="x-default"><link data-react-helmet="true" rel="preconnect" href="https://BH4D9OD16A-dsn.algolia.net" crossorigin="anonymous"><link rel="stylesheet" href="/warp/assets/css/styles.57c5b218.css">
<link rel="preload" href="/warp/assets/js/runtime~main.d7a12008.js" as="script">
<link rel="preload" href="/warp/assets/js/main.7a5a9da0.js" as="script">
</head>
<body>
<script>!function(){function t(t){document.documentElement.setAttribute("data-theme",t)}var e=function(){var t=null;try{t=localStorage.getItem("theme")}catch(t){}return t}();t(null!==e?e:"light")}()</script><div id="__docusaurus">
<div><a href="#" class="skipToContent_1oUP">Skip to main content</a></div><nav class="navbar navbar--fixed-top"><div class="navbar__inner"><div class="navbar__items"><button aria-label="Navigation bar toggle" class="navbar__toggle clean-btn" type="button" tabindex="0"><svg width="30" height="30" viewBox="0 0 30 30" aria-hidden="true"><path stroke="currentColor" stroke-linecap="round" stroke-miterlimit="10" stroke-width="2" d="M4 7h22M4 15h22M4 23h22"></path></svg></button><a class="navbar__brand" href="/warp/"><b class="navbar__title">WARP Pipelines</b></a></div><div class="navbar__items navbar__items--right"><a class="navbar__item navbar__link navbar__link--active" href="/warp/docs/get-started">Documentation</a><a class="navbar__item navbar__link" href="/warp/blog">Blog</a><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="navbar__item navbar__link header-github-link" aria-label="GitHub repository"></a><div class="react-toggle toggle_3Zt9 react-toggle--disabled"><div class="react-toggle-track" role="button" tabindex="-1"><div class="react-toggle-track-check"><span class="toggle_71bT">ðŸŒœ</span></div><div class="react-toggle-track-x"><span class="toggle_71bT">ðŸŒž</span></div><div class="react-toggle-thumb"></div></div><input type="checkbox" class="react-toggle-screenreader-only" aria-label="Switch between dark and light mode"></div><div class="searchBox_1Doo"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><svg width="20" height="20" class="DocSearch-Search-Icon" viewBox="0 0 20 20"><path d="M14.386 14.386l4.0877 4.0877-4.0877-4.0877c-2.9418 2.9419-7.7115 2.9419-10.6533 0-2.9419-2.9418-2.9419-7.7115 0-10.6533 2.9418-2.9419 7.7115-2.9419 10.6533 0 2.9419 2.9418 2.9419 7.7115 0 10.6533z" stroke="currentColor" fill="none" fill-rule="evenodd" stroke-linecap="round" stroke-linejoin="round"></path></svg><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"></span></button></div></div></div><div role="presentation" class="navbar-sidebar__backdrop"></div></nav><div class="main-wrapper docs-wrapper docs-doc-page"><div class="docPage_31aa"><button class="clean-btn backToTopButton_35hR" type="button"><svg viewBox="0 0 24 24" width="28"><path d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6z" fill="currentColor"></path></svg></button><aside class="docSidebarContainer_3Kbt"><div class="sidebar_15mo"><nav class="menu thin-scrollbar menu_Bmed menuWithAnnouncementBar_2WvA"><ul class="menu__list"><li class="menu__list-item"><a class="menu__link" href="/warp/docs/get-started">Welcome to WARP</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">About WARP</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#">WARP Pipelines</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">CEMBA</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Exome Germline Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">GDC Whole Genome Somatic Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Illumina Genotyping Array</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Imputation</a></li><li class="menu__list-item"><a class="menu__link menu__link--sublist menu__link--active" href="#" tabindex="0">Multiome scATAC and GEX</a><ul style="display:block;overflow:visible;height:auto" class="menu__list"><li class="menu__list-item"><a class="menu__link menu__link--active" aria-current="page" tabindex="0" href="/warp/docs/Pipelines/Multiome_Pipeline/README">Multiome Overview</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Optimus</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">RNA with UMIs</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Single Cell ATAC</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Slide-seq</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Nucleus Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Multi-Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Smart-seq2 Single Sample</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Ultima Genomics Whole Genome Germline</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#" tabindex="0">Whole Genome Germline Single Sample</a></li></ul></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">WDL Best Practices</a></li><li class="menu__list-item menu__list-item--collapsed"><a class="menu__link menu__link--sublist" href="#">Contribution Guide</a></li></ul></nav><button type="button" title="Collapse sidebar" aria-label="Collapse sidebar" class="button button--secondary button--outline collapseSidebarButton_1CGd"><svg width="20" height="20" aria-hidden="true" class="collapseSidebarButtonIcon_3E-R"><g fill="#7a7a7a"><path d="M9.992 10.023c0 .2-.062.399-.172.547l-4.996 7.492a.982.982 0 01-.828.454H1c-.55 0-1-.453-1-1 0-.2.059-.403.168-.551l4.629-6.942L.168 3.078A.939.939 0 010 2.528c0-.548.45-.997 1-.997h2.996c.352 0 .649.18.828.45L9.82 9.472c.11.148.172.347.172.55zm0 0"></path><path d="M19.98 10.023c0 .2-.058.399-.168.547l-4.996 7.492a.987.987 0 01-.828.454h-3c-.547 0-.996-.453-.996-1 0-.2.059-.403.168-.551l4.625-6.942-4.625-6.945a.939.939 0 01-.168-.55 1 1 0 01.996-.997h3c.348 0 .649.18.828.45l4.996 7.492c.11.148.168.347.168.55zm0 0"></path></g></svg></button></div></aside><main class="docMainContainer_3ufF"><div class="container padding-top--md padding-bottom--lg"><div class="row"><div class="col docItemCol_3FnS"><div class="docItemContainer_33ec"><article><div class="tocCollapsible_1PrD tocMobile_3Hoh"><button type="button" class="clean-btn tocCollapsibleButton_2O1e">On this page</button></div><div class="markdown"><header><h1>Multiome Overview</h1></header><table><thead><tr><th align="center">Pipeline Version</th><th align="center">Date Updated</th><th align="center">Documentation Author</th><th align="center">Questions or Feedback</th></tr></thead><tbody><tr><td align="center"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">Multiome v1.0.0</a></td><td align="center">June, 2023</td><td align="center">Kaylee Mathews</td><td align="center">Please file GitHub issues in warp or contact the <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">WARP Pipeline Development team</a></td></tr></tbody></table><p><img alt="Multiome_diagram" src="/warp/assets/images/multiome_diagram-9f63b411312ab30ae8d54dca539a30fa.png"></p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="introduction-to-the-multiome-workflow"></a>Introduction to the Multiome workflow<a class="hash-link" href="#introduction-to-the-multiome-workflow" title="Direct link to heading">#</a></h2><p>Multiome is an open-source, cloud-optimized pipeline developed in collaboration with members of the <a href="https://braininitiative.nih.gov/" target="_blank" rel="noopener noreferrer">BRAIN Initiative</a> (BICCN and <a href="https://brainblog.nih.gov/brain-blog/brain-issues-suite-funding-opportunities-advance-brain-cell-atlases-through-centers" target="_blank" rel="noopener noreferrer">BICAN</a> Sequencing Working Group) and <a href="https://nida.nih.gov/about-nida/organization/divisions/division-neuroscience-behavior-dnb/basic-research-hiv-substance-use-disorder/scorch-program" target="_blank" rel="noopener noreferrer">SCORCH</a> (see <a href="#acknowledgements">Acknowledgements</a> below). It supports the processing of 10x 3&#x27; single-cell and single-nucleus gene expression (GEX) and chromatin accessibility (ATAC) data generated with the <a href="https://www.10xgenomics.com/products/single-cell-multiome-atac-plus-gene-expression" target="_blank" rel="noopener noreferrer">10x Genomics Multiome assay</a>.</p><p>The workflow is a wrapper WDL script that calls two subworkflows: the <a href="/warp/docs/Pipelines/Optimus_Pipeline/README">Optimus workflow</a> for single-cell GEX data and the <a href="/warp/docs/Pipelines/ATAC/README">ATAC workflow</a> for single-cell ATAC data. </p><p>The GEX component corrects cell barcodes (CBs) and Unique Molecular Identifiers (UMIs), aligns reads to the genome, calculates per-barcode and per-gene quality metrics, and produces a raw cell-by-gene count matrix. </p><p>The ATAC component corrects CBs, aligns reads to the genome, calculates per-barcode quality metrics, and produces a fragment file.</p><p>The wrapper WDL is available in the WARP repository (see the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/Multiome.wdl" target="_blank" rel="noopener noreferrer">code here</a>).</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="quickstart-table"></a>Quickstart table<a class="hash-link" href="#quickstart-table" title="Direct link to heading">#</a></h2><p>The following table provides a quick glance at the Multiome pipeline features:</p><table><thead><tr><th>Pipeline features</th><th>Description</th><th>Source</th></tr></thead><tbody><tr><td>Assay type</td><td>10x single cell or single nucleus gene expression (GEX) and ATAC</td><td><a href="https://www.10xgenomics.com" target="_blank" rel="noopener noreferrer">10x Genomics</a></td></tr><tr><td>Overall workflow</td><td>Barcode correction, read alignment, gene and fragment quanitification</td><td></td></tr><tr><td>Workflow language</td><td>WDL 1.0</td><td><a href="https://github.com/openwdl/wdl" target="_blank" rel="noopener noreferrer">openWDL</a></td></tr><tr><td>Genomic Reference Sequence</td><td>GRCh38 human genome primary sequence</td><td>GENCODE <a href="https://www.gencodegenes.org/human/release_43.html" target="_blank" rel="noopener noreferrer">human reference files</a></td></tr><tr><td>Gene annotation reference (GTF)</td><td>Reference containing gene annotations</td><td>GENCODE <a href="https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_43/gencode.v43.annotation.gtf.gz" target="_blank" rel="noopener noreferrer">human GTF</a></td></tr><tr><td>Aligners</td><td>STARsolo (GEX), BWA-mem2 (ATAC)</td><td><a href="https://www.biorxiv.org/content/10.1101/2021.05.05.442755v1" target="_blank" rel="noopener noreferrer">Kaminow et al. 2021</a>, <a href="https://ieeexplore.ieee.org/document/8820962" target="_blank" rel="noopener noreferrer">Vasimuddin et al. 2019</a></td></tr><tr><td>Transcript and fragment quantification</td><td>STARsolo (GEX), SnapATAC2 (ATAC)</td><td><a href="https://www.biorxiv.org/content/10.1101/2021.05.05.442755v1" target="_blank" rel="noopener noreferrer">Kaminow et al. 2021</a>, <a href="https://kzhang.org/SnapATAC2/" target="_blank" rel="noopener noreferrer">SnapATAC2</a></td></tr><tr><td>Data input file format</td><td>File format in which sequencing data is provided</td><td><a href="https://academic.oup.com/nar/article/38/6/1767/3112533" target="_blank" rel="noopener noreferrer">FASTQ</a></td></tr><tr><td>Data output file format</td><td>File formats in which Multiome output is provided</td><td><a href="http://samtools.github.io/hts-specs/" target="_blank" rel="noopener noreferrer">BAM</a> and <a href="https://anndata.readthedocs.io/en/latest/" target="_blank" rel="noopener noreferrer">h5ad</a></td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="set-up"></a>Set-up<a class="hash-link" href="#set-up" title="Direct link to heading">#</a></h2><h3><a aria-hidden="true" tabindex="-1" class="anchor anchor__h3 anchorWithStickyNavbar_31ik" id="multiome-installation"></a>Multiome installation<a class="hash-link" href="#multiome-installation" title="Direct link to heading">#</a></h3><p>To download the latest Multiome release, see the release tags prefixed with &quot;Multiome&quot; on the WARP <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">releases page</a>. All Multiome pipeline releases are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/Multiome.changelog.md" target="_blank" rel="noopener noreferrer">Multiome changelog</a>. </p><p>To discover and search releases, use the WARP command-line tool <a href="https://github.com/broadinstitute/warp/tree/master/wreleaser" target="_blank" rel="noopener noreferrer">Wreleaser</a>.</p><p>If youâ€™re running a Multiome workflow version prior to the latest release, the accompanying documentation for that release may be downloaded with the source code on the WARP <a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer">releases page</a> (see the source code folder).</p><p>Multiome can be deployed using <a href="https://cromwell.readthedocs.io/en/stable/" target="_blank" rel="noopener noreferrer">Cromwell</a>, a GA4GH compliant, flexible workflow management system that supports multiple computing platforms. The workflow can also be run in <a href="https://app.terra.bio" target="_blank" rel="noopener noreferrer">Terra</a>, a cloud-based analysis platform. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="inputs"></a>Inputs<a class="hash-link" href="#inputs" title="Direct link to heading">#</a></h2><table><thead><tr><th>Input name</th><th>Description</th><th>Type</th></tr></thead><tbody><tr><td>input_id</td><td>Unique identifier describing the biological sample or replicate that corresponds with the FASTQ files; can be a human-readable name or UUID.</td><td>String</td></tr><tr><td>annotations_gtf</td><td>GTF file containing gene annotations used for GEX cell metric calculation and ATAC fragment metrics; must match the GTF used to build the STAR aligner.</td><td>File</td></tr><tr><td>gex_r1_fastq</td><td>Array of read 1 FASTQ files representing a single GEX 10x library.</td><td>Array[File]</td></tr><tr><td>gex_r2_fastq</td><td>Array of read 2 FASTQ files representing a single GEX 10x library.</td><td>Array[File]</td></tr><tr><td>gex_i1_fastq</td><td>Optional array of index FASTQ files representing a single GEX 10x library; multiplexed samples are not currently supported, but the file may be passed to the pipeline.</td><td>Array[File]</td></tr><tr><td>tar_star_reference</td><td>TAR file containing a species-specific reference genome and GTF for Optimus (GEX) pipeline.</td><td>File</td></tr><tr><td>ref_genome_fasta</td><td>Genome FASTA file used for building the indices.</td><td>File</td></tr><tr><td>mt_genes</td><td>Optional file for the Optimus (GEX) pipeline containing mitochondrial gene names used for metric calculation; default assumes &#x27;mt&#x27; prefix in GTF (case insensitive).</td><td>File</td></tr><tr><td>mt_sequence</td><td>Optional string for the Optimus (GEX) pipeline specifying the chromosome listed in the GTF that contains mitrochondrial genes. For example, when using human and mouse GENCODE GTFs, this should be set to &quot;chrM&quot;. If no mt_sequence is provided, Dropseq metrics will not run.</td><td>String</td></tr><tr><td>counting_mode</td><td>Optional string that determines whether the Optimus (GEX) pipeline should be run in single-cell mode (sc_rna) or single-nucleus mode (sn_rna); default is &quot;sn_rna&quot;.</td><td>String</td></tr><tr><td>tenx_chemistry_version</td><td>Optional integer for the Optimus (GEX) pipeline specifying the 10x version chemistry the data was generated with; validated by examination of the first read 1 FASTQ file read structure; default is &quot;3&quot;.</td><td>Integer</td></tr><tr><td>emptydrops_lower</td><td>Optional threshold for UMIs for the Optimus (GEX) pipeline that empty drops tool should consider for determining cell; data below threshold is not removed; default is &quot;100&quot;.</td><td>Integer</td></tr><tr><td>force_no_check</td><td>Optional boolean for the Optimus (GEX) pipeline indicating if the pipeline should perform checks; default is &quot;false&quot;.</td><td>Boolean</td></tr><tr><td>ignore_r1_read_length</td><td>Optional boolean for the Optimus (GEX) pipeline indicating if the pipeline should ignore barcode chemistry check; if &quot;true&quot;, the workflow will not ensure the <code>10x_chemistry_version</code> input matches the chemistry in the read 1 FASTQ; default is &quot;false&quot;.</td><td>Boolean</td></tr><tr><td>star_strand_mode</td><td>Optional string for the Optimus (GEX) pipeline for performing STARsolo alignment on forward stranded, reverse stranded, or unstranded data; default is &quot;Forward&quot;.</td><td>String</td></tr><tr><td>count_exons</td><td>Optional boolean for the Optimus (GEX) pipeline indicating if the workflow should calculate exon counts <strong>when in single-nucleus (sn_rna) mode</strong>; if &quot;true&quot; in sc_rna mode, the workflow will return an error; default is &quot;false&quot;.</td><td>Boolean</td></tr><tr><td>gex_whitelist</td><td>Optional file containing the list of valid barcodes for 10x multiome GEX data; default is &quot;gs://broad-gotc-test-storage/Multiome/input/737K-arc-v1_gex.txt&quot;.</td><td>File</td></tr><tr><td>atac_r1_fastq</td><td>Array of read 1 paired-end FASTQ files representing a single 10x multiome ATAC library.</td><td>Array[File]</td></tr><tr><td>atac_r2_fastq</td><td>Array of barcodes FASTQ files representing a single 10x multiome ATAC library.</td><td>Array[File]</td></tr><tr><td>atac_r3_fastq</td><td>Array of read 2 paired-end FASTQ files representing a single 10x multiome ATAC library.</td><td>Array[File]</td></tr><tr><td>tar_bwa_reference</td><td>TAR file containing the reference index files for BWA-mem alignment for the ATAC pipeline .</td><td>File</td></tr><tr><td>chrom_sizes</td><td>File containing the genome chromosome sizes; used to calculate ATAC fragment file metrics.</td><td>File</td></tr><tr><td>adapter_seq_read1</td><td>Optional string describing the adapter sequence for ATAC read 1 paired-end reads to be used during adapter trimming with Cutadapt; default is &quot;GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG&quot;.</td><td>String</td></tr><tr><td>adapter_seq_read3</td><td>Optional string describing the adapter sequence for ATAC read 2 paired-end reads to be used during adapter trimming with Cutadapt; default is &quot;TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG&quot;.</td><td>String</td></tr><tr><td>atac_whitelist</td><td>Optional file containing the list of valid barcodes for 10x multiome ATAC adata; default is &quot;gs://broad-gotc-test-storage/Multiome/input/737K-arc-v1_atac.txt&quot;.</td><td>File</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="tasks"></a>Tasks<a class="hash-link" href="#tasks" title="Direct link to heading">#</a></h2><p>The Multiome workflow calls two subworkflows, which are described briefly in the table below. For more details on each subworkflow, including the tasks that they call, see the documentation linked in the table.</p><table><thead><tr><th>Subworkflow</th><th>Software</th><th>Description</th></tr></thead><tbody><tr><td>ATAC (<a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/atac.wdl" target="_blank" rel="noopener noreferrer">WDL</a> and <a href="/warp/docs/Pipelines/ATAC/README">documentation</a>)</td><td>fastqprocess, bwa-mem, SnapATAC2</td><td>Workflow used to analyze 10x single-cell ATAC data.</td></tr><tr><td>Optimus (<a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/optimus/Optimus.wdl" target="_blank" rel="noopener noreferrer">WDL</a> and <a href="/warp/docs/Pipelines/Optimus_Pipeline/README">documentation</a>)</td><td>fastqprocess, STARsolo, Emptydrops</td><td>Workflow used to analyze 10x single-cell GEX data.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="outputs"></a>Outputs<a class="hash-link" href="#outputs" title="Direct link to heading">#</a></h2><table><thead><tr><th>Output variable name</th><th>Filename, if applicable</th><th>Output format and description</th></tr></thead><tbody><tr><td>bam_aligned_output_atac</td><td><code>&lt;input_id&gt;_atac.bam</code></td><td>BAM file containing aligned reads from ATAC workflow.</td></tr><tr><td>fragment_file_atac</td><td><code>&lt;input_id&gt;_atac.fragments.tsv</code></td><td>TSV file containing fragment start and stop coordinates per barcode.</td></tr><tr><td>snap_metrics_atac</td><td><code>&lt;input_id&gt;_atac.metrics.h5ad</code></td><td>h5ad (Anndata) file containing per-barcode metrics from SnapATAC2.</td></tr><tr><td>pipeline_version_out</td><td>N.A.</td><td>String describing the Optimus pipeline version used.</td></tr><tr><td>genomic_reference_version_gex</td><td><code>&lt;reference_version&gt;.txt</code></td><td>File containing the Genome build, source and GTF annotation version.</td></tr><tr><td>bam_gex</td><td><code>&lt;input_id&gt;_gex.bam</code></td><td>BAM file containing aligned reads from Optimus workflow.</td></tr><tr><td>matrix_gex</td><td><code>&lt;input_id&gt;_gex_sparse_counts.npz</code></td><td>NPZ file containing raw gene by cell counts.</td></tr><tr><td>matrix_row_index_gex</td><td><code>&lt;input_id&gt;_gex_sparse_counts_row_index.npy</code></td><td>NPY file containing the row indices.</td></tr><tr><td>matrix_col_index_gex</td><td><code>&lt;input_id&gt;_gex_sparse_counts_col_index.npy</code></td><td>NPY file containing the column indices.</td></tr><tr><td>cell_metrics_gex</td><td><code>&lt;input_id&gt;_gex.cell_metrics.csv.gz</code></td><td>CSV file containing the per-cell (barcode) metrics.</td></tr><tr><td>gene_metrics_gex</td><td><code>&lt;input_id&gt;_gex.gene_metrics.csv.gz</code></td><td>CSV file containing the per-gene metrics.</td></tr><tr><td>cell_calls_gex</td><td><code>&lt;input_id&gt;_gex.emptyDrops</code></td><td>TSV file containing the EmptyDrops results when the Optimus workflow is run in sc_rna mode.</td></tr><tr><td>picard_metrics_gex</td><td><code>&lt;input_id&gt;_gex.tsv</code></td><td>Cell barcode-specific metrics generated using Dropseq tool&#x27;s <a href="https://github.com/broadinstitute/Drop-seq/blob/master/src/java/org/broadinstitute/dropseqrna/barnyard/SingleCellRnaSeqMetricsCollector.java" target="_blank" rel="noopener noreferrer">SingleCellRnaSeqMetricsCollector</a>.</td></tr><tr><td>h5ad_output_file_gex</td><td><code>&lt;input_id&gt;_gex.h5ad</code></td><td>h5ad (Anndata) file containing the raw cell-by-gene count matrix, gene metrics, cell metrics, and global attributes.</td></tr></tbody></table><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="versioning-and-testing"></a>Versioning and testing<a class="hash-link" href="#versioning-and-testing" title="Direct link to heading">#</a></h2><p>All Multiome pipeline releases are documented in the <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/Multiome.changelog.md" target="_blank" rel="noopener noreferrer">Multiome changelog</a> and tested using <a href="https://github.com/broadinstitute/warp/blob/develop/pipelines/skylab/multiome/test_inputs/test_data_overview.md" target="_blank" rel="noopener noreferrer">plumbing and scientific test data</a>. To learn more about WARP pipeline testing, see <a href="https://broadinstitute.github.io/warp/docs/About_WARP/TestingPipelines" target="_blank" rel="noopener noreferrer">Testing Pipelines</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="consortia-support"></a>Consortia support<a class="hash-link" href="#consortia-support" title="Direct link to heading">#</a></h2><p>This pipeline is supported by the <a href="https://braininitiative.nih.gov/" target="_blank" rel="noopener noreferrer">BRAIN Initiative</a> (BICCN and BICAN). </p><p>If your organization also uses this pipeline, we would like to list you! Please reach out to us by contacting the <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">WARP Pipeline Development team</a>.</p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="acknowledgements"></a>Acknowledgements<a class="hash-link" href="#acknowledgements" title="Direct link to heading">#</a></h2><p>We are immensely grateful to the members of the BRAIN Initiative (BICAN Sequencing Working Group) and SCORCH for their invaluable and exceptional contributions to this pipeline. Our heartfelt appreciation goes to Alex Dobin, Aparna Bhaduri, Alec Wysoker, Anish Chakka, Brian Herb, Daofeng Li, Fenna Krienen, Guo-Long Zuo, Jeff Goldy, Kai Zhang, Khalid Shakir, Bo Li, Mariano Gabitto, Michael DeBerardine, Mengyi Song, Melissa Goldman, Nelson Johansen, James Nemesh, and Theresa Hodges for their unwavering dedication and remarkable efforts. </p><h2><a aria-hidden="true" tabindex="-1" class="anchor anchor__h2 anchorWithStickyNavbar_31ik" id="feedback"></a>Feedback<a class="hash-link" href="#feedback" title="Direct link to heading">#</a></h2><p>Please help us make our tools better by contacting the <a href="mailto:warp-pipelines-help@broadinstitute.org" target="_blank" rel="noopener noreferrer">WARP Pipeline Development team</a> for pipeline-related suggestions or questions.</p></div><footer class="docusaurus-mt-lg"><div class="row"><div class="col"><a href="https://github.com/broadinstitute/warp/edit/develop/website/docs/Pipelines/Multiome_Pipeline/README.md" target="_blank" rel="noreferrer noopener"><svg fill="currentColor" height="20" width="20" viewBox="0 0 40 40" class="iconEdit_2_ui" aria-hidden="true"><g><path d="m34.5 11.7l-3 3.1-6.3-6.3 3.1-3q0.5-0.5 1.2-0.5t1.1 0.5l3.9 3.9q0.5 0.4 0.5 1.1t-0.5 1.2z m-29.5 17.1l18.4-18.5 6.3 6.3-18.4 18.4h-6.3v-6.2z"></path></g></svg>Edit this page</a></div><div class="col lastUpdated_13-_">Last updated on <b><time datetime="2023-06-26T14:14:29.000Z">6/26/2023</time></b> by <b>ekiernan</b></div></div></footer></article><nav class="pagination-nav docusaurus-mt-lg" aria-label="Docs pages navigation"><div class="pagination-nav__item"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Imputation_Pipeline/references_overview"><div class="pagination-nav__sublabel">Previous</div><div class="pagination-nav__label">Â« Creating the Imputation Pipeline&#x27;s Modified 1000 Genomes Reference</div></a></div><div class="pagination-nav__item pagination-nav__item--next"><a class="pagination-nav__link" href="/warp/docs/Pipelines/Optimus_Pipeline/README"><div class="pagination-nav__sublabel">Next</div><div class="pagination-nav__label">Optimus Overview Â»</div></a></div></nav></div></div><div class="col col--3"><div class="tableOfContents_35-E thin-scrollbar"><ul class="table-of-contents table-of-contents__left-border"><li><a href="#introduction-to-the-multiome-workflow" class="table-of-contents__link">Introduction to the Multiome workflow</a></li><li><a href="#quickstart-table" class="table-of-contents__link">Quickstart table</a></li><li><a href="#set-up" class="table-of-contents__link">Set-up</a><ul><li><a href="#multiome-installation" class="table-of-contents__link">Multiome installation</a></li></ul></li><li><a href="#inputs" class="table-of-contents__link">Inputs</a></li><li><a href="#tasks" class="table-of-contents__link">Tasks</a></li><li><a href="#outputs" class="table-of-contents__link">Outputs</a></li><li><a href="#versioning-and-testing" class="table-of-contents__link">Versioning and testing</a></li><li><a href="#consortia-support" class="table-of-contents__link">Consortia support</a></li><li><a href="#acknowledgements" class="table-of-contents__link">Acknowledgements</a></li><li><a href="#feedback" class="table-of-contents__link">Feedback</a></li></ul></div></div></div></div></main></div></div><footer class="footer footer--dark"><div class="container"><div class="row footer__links"><div class="col footer__col"><div class="footer__title">Docs</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/get-started">Get Started</a></li></ul></div><div class="col footer__col"><div class="footer__title">Guide and Policy</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/docs/contribution/README">Contribution Guide</a></li><li class="footer__item"><a class="footer__link-item" href="/warp/privacy">Privacy</a></li></ul></div><div class="col footer__col"><div class="footer__title">Resources</div><ul class="footer__items"><li class="footer__item"><a class="footer__link-item" href="/warp/blog">Blog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp/releases" target="_blank" rel="noopener noreferrer" class="footer__link-item">Changelog</a></li><li class="footer__item"><a href="https://github.com/broadinstitute/warp" target="_blank" rel="noopener noreferrer" class="footer__link-item"><span>GitHub<svg width="13.5" height="13.5" aria-hidden="true" viewBox="0 0 24 24" class="iconExternalLink_3J9K"><path fill="currentColor" d="M21 13v10h-21v-19h12v2h-10v15h17v-8h2zm3-12h-10.988l4.035 4-6.977 7.07 2.828 2.828 6.977-7.07 4.125 4.172v-11z"></path></svg></span></a></li></ul></div></div><div class="footer__bottom text--center"><div class="footer__copyright">Copyright Â© 2023 Copyright Â© Data Sciences Platform, Broad Institute.</div></div></div></footer></div>
<script src="/warp/assets/js/runtime~main.d7a12008.js"></script>
<script src="/warp/assets/js/main.7a5a9da0.js"></script>
</body>
</html>